### PR TITLE
[VEGA-1063] Adding new global cache key

### PR DESCRIPTION
**Pull request by OLingard**
## Changes:

- Moving to the v2 API endpoints
- Changing the tests to use static files so it's easier to read
- Adding the global cache key capability

### CODE REVIEW

File: docs/resources/mte_config.md [hunk 1]
```diff
@@ -62,24 +76,40 @@ Required:
...
+### Nested Schema for `config.cache`
+
+Optional:
+
+- `keys` (Attributes) An object specifying header and cookie names which should be added to the cache key. The result of this would lead to separate cache hits for requests with different values of the header or cookie. One of this (see [below for nested schema](#nestedatt--config--cache--keys))
+- `path_rules` (Attributes) A set of glob rules which identify when the cache settings should be activated. (see [below for nested schema](#nestedatt--config--cache--path_rules))
+- `ttl_seconds` (Number) An integer that will be used to specify the time that the response of the route should be stored in the cache, in seconds.
+
+<a id="nestedatt--config--cache--keys"></a>
+### Nested Schema for `config.cache.keys`
```
Comment (JosephMcDonagh) [inline docs/resources/mte_config.md:101]: Can we put required above optional pls

File: internal/provider/config_resource.go [hunk 2]
```diff
@@ -99,6 +110,28 @@ func (m *MTEConfigResource) Configure(ctx context.Context, req resource.Configur
 	m.client = resourceData.client
 }
 
+func (m *MTEConfigResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
```
Comment (JosephMcDonagh) [inline internal/provider/config_resource.go:230]: What does this do in the flow? Is this not done automatically via the schema?

File: internal/provider/config_resource.go [hunk 3]
```diff
@@ -368,42 +407,60 @@ func (m *MTEConfigResourceModel) transformToApiRequestBody() client.MTEConfigDto
...
+					noneMatch[i] = c.ValueString()
+				}
+				cacheBody.PathRules = &client.MatcherDto{
+					AnyMatch:  anyMatch,
+					NoneMatch: noneMatch,
+				}
+			}
+			httpCache[i] = cacheBody
+		}
+		dto.Cache = httpCache
```
Comment (JosephMcDonagh) [inline internal/provider/config_resource.go]: Maybe this (and its equivalent in the other transformation) should be moved into their own functions, to stop the transformation functions getting too chunky

File: internal/provider/config_resource.go (hunk 2 above)
Comment (OLingard) [reply internal/provider/config_resource.go:230]: This one is used for when there are two optional fields where one of them needs to be present (so an either/or scenario), you can't just do that through the optional or required flag, additional logic is needed

File: docs/resources/mte_config.md (hunk 1 above)
Comment (OLingard) [reply docs/resources/mte_config.md:101]: This is generated by Go, the reason for the layout is that the config.cache.keys is nested within the optional config.cache

File: internal/provider/config_resource.go (hunk 3 above)
Comment (OLingard) [reply internal/provider/config_resource.go]: Good shout, updated

File: docs/resources/mte_config.md (hunk 1 above)
Comment (JosephMcDonagh) [reply docs/resources/mte_config.md:101]: makes sense

**Review by JosephMcDonagh — APPROVED** [review]
LG2M

**Pull request merged**